main.js

```diff
@@ -0,0 +1,132 @@
+// console.log(GET_MIN_NUM("k", 7, 3, 5)) // You have to put only digits!
+console.log(GET_MIN_NUM()) // Not enough digits!
+
+
+// 2. Написати функцію, яка приймає 2 аргументи та шукає найбільший спільний дільник для них. Якщо такого числа немає, то повідомляти про відсутність
+const GET_GREATEST_COMMON_DENOMINATOR = function (num1, num2) {
+    if (num1 === undefined || num2 === undefined || num1 == null || num2 == null) {
+        return "Not enough digits!"
+    } else if ((typeof num1 !== 'number') || (typeof num2 !== 'number')) {
+        return "You have to put only digits!";
+    } else if (arguments.length > 2) {
+        return "You have to put only 2 digits, not more!";
+    }
+      num1 = Math.abs(num1);
+      num2 = Math.abs(num2);
+      while(num2) {
+        const NUM3 = num2;
+        num2 = num1 % num2;
+        num1 = NUM3;
+      }
+      return `${num1} is the greatest common denominator!`;
+    }
+  console.log(GET_GREATEST_COMMON_DENOMINATOR(8, 16)) // 8 is the greatest common denominator!
+//   console.log(GET_GREATEST_COMMON_DENOMINATOR(8, 16, 28)) // You have to put only 2 digits, not more!
+//   console.log(GET_GREATEST_COMMON_DENOMINATOR(16)) // Not enough digits!
+//   console.log(GET_GREATEST_COMMON_DENOMINATOR(8, "k")) // You have to put only digits!
+  
+  // 3. Написати функцію, яка приймає 1 аргумент та перевіряє чи дане число ідеальне
+  const PERFECT_NUMBER = function (num) {
+    let step = 0;
+    for (let i = 1; i <= num/2; i++) {
+        if (num % i === 0) {
+            step += i;
+        }
+    }
+    if (num === undefined || num == null) {
+        return "Not enough digits!"
+    } else if (arguments.length > 1) {
+        return "You have to put only 1 digit, not more!";
+    } else if ((typeof num !== 'number')) {
+        return "You have to put only digits!";
+    } else if (step === num && step !== 0) {
+        return `The number ${num} is perfect!`;
+    } else {
+        return `The number ${num} is NOT perfect!`;
+    }
+  }
+  console.log(PERFECT_NUMBER(6)) // The number 6 is perfect!
+  //console.log(PERFECT_NUMBER(5)) // The number 5 is NOT perfect!
+  //console.log(PERFECT_NUMBER(5, 7)) // You have to put only 1 digit, not more!
+  //console.log(PERFECT_NUMBER()) // Not enough digits!
+  //console.log(PERFECT_NUMBER("g")) // You have to put only digits!
+
+  // 4. Написати функцію, яка приймає 2 аргументи та обраховує суму в цьому проміжку. Зверніть увагу, що можна передати 10 та -2, як аргументи. Суму всіх чисел в цьому проміжку також треба обчислити
+    const RANGE_SUM = function(num1, num2) {
+    let sum = 0;
+    if (num1 === undefined || num2 === undefined || num1 == null || num2 == null) {
+        return "Not enough digits!"
+    } else if ((typeof num1 !== 'number') || (typeof num2 !== 'number')) {
+        return "You have to put only digits!";
+    } else if (arguments.length > 2) {
+        return "You have to put only 2 digits, not more!";
```

Comment:
This code repeats a lot in the whole JS. Consider making a function and to call it whenever need it. If there is something in your code that repeats - automatically it is the best solution to wrap it inside function